crates/duty-tracker/src/contract_manager.rs

```diff
@@ -301,11 +325,20 @@ impl ContractManager {
                         }
                     },
 
-                    Some(block) = block_sub.next() => {
-                        let blockhash = block.block.block_hash();
-                        let block_height = block.block.bip34_block_height().expect("must have valid height");
-                        info!(%blockhash, %block_height, "processing block");
-                        match ctx.process_block(block.block).await {
+                    // Then prioritize processing the backlog of chain state.
+                    Some(block_event) = block_sub.next() => {
+                        if block_event.status != BlockStatus::Buried {
+                            continue;
+                        }
+
+                        let blockhash = block_event.block.block_hash();
+                        let block_height = block_event.block.bip34_block_height().expect("must have valid height");
+                        info!(%block_height, %blockhash, "processing block");
```

Comment:
can you make a similar change in [btc-notify](https://github.com/alpenlabs/strata-bridge/blob/6ae2f51e405a2bf261acf8b122b93e3ac87cae73/crates/btc-notify/src/state_machine.rs#L131)?
and I'm guessing it would be useful to add the height [here](https://github.com/alpenlabs/strata-bridge/blob/6ae2f51e405a2bf261acf8b122b93e3ac87cae73/crates/btc-notify/src/client.rs#L143), though generally I'm not sure if we need this and the previous one, but let's keep for sake of debugging.